__tests__/app.test.js

```diff
@@ -31,9 +33,560 @@ describe("NCNews-Server Unit Tests", () => {
+            // include thorough sql injection tests in each section below
+
+            describe(`GET /api/articles/?topic=`, () => {
+
+                test(`[ 200 ] When passed "topic=cats" as a url parameter, responds with the 1 cat topic article`, () => {
+
+                    return request(app).get(`/api/articles?topic=cats`).expect(200).then((response) => {
+                        const articlesArray = response.body.articles;   
+                        expect(articlesArray).toHaveLength(1);
+                        expect(articlesArray[0]["topic"]).toBe("cats");
+        
+                    });
+                });
+
+                test(`[ 200 ] A topic with zero matches returns an empty array`, () => {
+
+                    return request(app).get(`/api/articles?topic=k`).expect(200).then((response) => {
+                        const articlesArray = response.body.articles;   
+                        expect(articlesArray).toHaveLength(0);
+                               
+                    });
+                });
+            });
+
+            describe(`GET /api/articles/?order=`, () => {
+
+                test(`[ 200 ] By default when passed no order param, server responds with an array of objects, where the objects are sorted by Date using the 'created_at' property in Descending order.`, () => {
+
+                    return request(app).get(`/api/articles`).expect(200).then((response) => {
+                        const articlesArray = response.body.articles;   
+                        expect(articlesArray).toHaveLength(12);
+        
+                        articlesArray.forEach((e, index) => {
+                            if (index < articlesArray.length - 1) {
+                            expect(Date.parse(articlesArray[index][`created_at`]) > Date.parse(articlesArray[index + 1][`created_at`])).toBe(true);
+                        }});
+                    });
+                });
+
+                test(`[ 200 ] When passed "asc" as a URL parameter, server responds with an array of objects, where the objects are sorted by Date using the 'created_at' property in Ascending order.`, () => {
+
+                    return request(app).get(`/api/articles?order=asc`).expect(200).then((response) => {
+                        const articlesArray = response.body.articles;   
+                        expect(articlesArray).toHaveLength(12);
+        
+                        articlesArray.forEach((e, index) => {
+                            if (index < articlesArray.length - 1) {
+                            expect(Date.parse(articlesArray[index][`created_at`]) < Date.parse(articlesArray[index + 1][`created_at`])).toBe(true);
+                        }});
+                    });
+                });
+
+                test(`[ 200 ] Makes sure this endpoint works if URL is formatted slightly differently`, () => {
+
+                    return request(app).get(`/api/articles/?order=asc`).expect(200).then((response) => {
```

Comment:
I imagine this test passed right away, so I think its redundant, but I suppose there's no harm in leaving it in